pkg/detector/preemption_test.go

```diff
@@ -0,0 +1,659 @@
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "foo",
+							Namespace: "test",
+							Labels: map[string]string{
+								policyv1alpha1.PropagationPolicyPermanentIDLabel: "policy-1",
+							},
+						},
+						Spec: policyv1alpha1.PropagationSpec{
+							Priority:   ptr.To[int32](3),
+							Preemption: policyv1alpha1.PreemptAlways,
+							ResourceSelectors: []policyv1alpha1.ResourceSelector{
+								{
+									APIVersion: "apps/v1",
+									Kind:       "Deployment",
+									Namespace:  "test",
+									Name:       "default",
+								},
+							},
+						},
+					},
+					&policyv1alpha1.PropagationPolicy{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "foo-2",
+							Namespace: "test",
+							Labels: map[string]string{
+								policyv1alpha1.PropagationPolicyPermanentIDLabel: "policy-2",
+							},
+						},
+						Spec: policyv1alpha1.PropagationSpec{
+							Priority:   ptr.To[int32](4),
+							Preemption: policyv1alpha1.PreemptAlways,
+							ResourceSelectors: []policyv1alpha1.ResourceSelector{
+								{
+									APIVersion: "apps/v1",
+									Kind:       "Deployment",
+									Namespace:  "test-2",
+									Name:       "default-2",
+								},
+							},
+						},
+					},
+				}
+				return fake.NewClientBuilder().WithScheme(scheme).WithObjects(obj...)
+			},
+			wantQueueSize: 2,
+		},
+		{
+			name: "no policy to preempt",
+			newPolicy: &policyv1alpha1.PropagationPolicy{
+				ObjectMeta: metav1.ObjectMeta{Name: "app", Namespace: "test"},
+				Spec: policyv1alpha1.PropagationSpec{
+					Priority: ptr.To[int32](2),
+					ResourceSelectors: []policyv1alpha1.ResourceSelector{
+						{
+							APIVersion:    "apps/v1",
+							Kind:          "Deployment",
+							Namespace:     "test",
+							LabelSelector: &metav1.LabelSelector{MatchLabels: map[string]string{"app": "nginx"}},
+						},
+					},
+				},
+			},
+			oldPolicy: &policyv1alpha1.PropagationPolicy{
+				ObjectMeta: metav1.ObjectMeta{Name: "app", Namespace: "test"},
+				Spec: policyv1alpha1.PropagationSpec{
+					Priority: ptr.To[int32](4),
+					ResourceSelectors: []policyv1alpha1.ResourceSelector{
+						{
+							APIVersion:    "apps/v1",
+							Kind:          "Deployment",
+							Namespace:     "test",
+							LabelSelector: &metav1.LabelSelector{MatchLabels: map[string]string{"app": "nginx"}},
+						},
+					},
+				},
+			},
+			objects: nil,
+			setupClient: func() *fake.ClientBuilder {
+				return fake.NewClientBuilder().WithScheme(scheme)
+			},
+			wantQueueSize: 0,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fakeClient := tt.setupClient().Build()
+			stopCh := make(chan struct{})
+			defer close(stopCh)
+			var fakeDynamicClient *dynamicfake.FakeDynamicClient
+			if tt.objects != nil {
+				fakeDynamicClient = dynamicfake.NewSimpleDynamicClient(scheme, tt.objects...)
+			} else {
+				fakeDynamicClient = dynamicfake.NewSimpleDynamicClient(scheme)
+			}
```

Comment:
It's okay to just write that: `fakeDynamicClient = dynamicfake.NewSimpleDynamicClient(scheme, tt.objects...)`